src/webxdc/integration.rs

```diff
@@ -0,0 +1,129 @@
+use crate::chat::{send_msg, ChatId};
+use crate::config::Config;
+use crate::contact::ContactId;
+use crate::context::Context;
+use crate::message::{Message, MsgId, Viewtype};
+use crate::param::Param;
+use crate::webxdc::{maps_integration, StatusUpdateItem, StatusUpdateSerial};
+use anyhow::Result;
+
+impl Context {
+    /// Set Webxdc file as integration.
+    pub async fn set_webxdc_integration(&self, file: String) -> Result<()> {
+        let chat_id = ChatId::create_for_contact(self, ContactId::SELF).await?;
+        let mut msg = Message::new(Viewtype::Webxdc);
+        msg.set_file(file.as_str(), None);
+        msg.hidden = true;
+        msg.param.set_int(Param::WebxdcIntegration, 1);
+        msg.param.set_int(Param::GuaranteeE2ee, 1); // needed to pass `internet_access` requirements
+        send_msg(self, chat_id, &mut msg).await?;
+        Ok(())
+    }
+
+    /// Get Webxdc instance used for optional integrations.
+    /// If there is no integration, the caller may decide to add a default one.
+    pub async fn init_webxdc_integration(
```

Comment:
This needs a documentation for `integrate_for` (maybe `integrate_in` btw?) and the function result

Reply:
i copy some of the cffi documentation here